templates/d/src/tic80.d

```diff
@@ -4,13 +4,55 @@ extern(C):
 
 struct MouseData {
     short x; short y; 
-    short scrollx; short scrolly;
+    byte scrollx; byte scrolly;
     bool left; bool middle; bool right;
 }
 
 const int WIDTH = 240;
 const int HEIGHT = 136;
 
+// These are pointers.
+const FRAMEBUFFER_PTR = cast(ubyte*)0;
+const TILES_PTR = cast(ubyte*)0x4000;
+const SPRITES_PTR = cast(ubyte*)0x6000;
+const MAP_PTR = cast(ubyte*)0x8000;
+const GAMEPADS_PTR = cast(ubyte*)0xFF80;
+const MOUSE_PTR = cast(ubyte*)0xFF84;
+const KEYBOARD_PTR = cast(ubyte*)0xFF88;
+const SFX_STATE_PTR = cast(ubyte*)0xFF8C;
+const SOUND_REGISTERS_PTR = cast(ubyte*)0xFF9C;
+const WAVEFORMS_PTR = cast(ubyte*)0xFFE4;
+const SFX_PTR = cast(ubyte*)0x100E4;
+const MUSIC_PATTERNS_PTR = cast(ubyte*)0x11164;
+const MUSIC_TRACKS_PTR = cast(ubyte*)0x13E64;
+const SOUND_STATE_PTR = cast(ubyte*)0x13FFC;
+const STEREO_VOLUME_PTR = cast(ubyte*)0x14000;
+const PERSISTENT_MEMORY_PTR = cast(ubyte*)0x14004;
+const SPRITE_FLAGS_PTR = cast(ubyte*)0x14404;
+const SYSTEM_FONT_PTR = cast(ubyte*)0x14604;
+const WASM_FREE_RAM_PTR = cast(ubyte*)0x18000;
```

Comment:
Do you really think we should provide this out of the box?  Does D not allocate globals and such things starting here automatically?

Reply:
I suppose you are referring to WASM_FREE_RAM_PTR? Since TIC-80 allows peeking and poking anywhere, I put this in as well. But yes, generally the game programmer will leave to D to manage memory.

Reply:
I know. I'm just not sure we want to _encourage_ this.  Someone who knows what they are doing can add that line to their own source easily enough without us including it in the core examples.   And if they do not know what they are doing they probably shouldn't be using `WASM_FREE_RAM_PTR` in the first place... 

Thoughts?

Reply:
Let's leave it out then.